catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -87,28 +89,54 @@ protected void onCreate(Bundle savedInstanceState) {
 		PreferenceManager.setDefaultValues(this, R.xml.preferences, true);
 		ScreenValueHandler.updateScreenWidthAndHeight(this);
 
-		int oldPrivacyPolicyHash = PreferenceManager.getDefaultSharedPreferences(this)
+		int oldPrivacyPolicy = PreferenceManager.getDefaultSharedPreferences(this)
 						.getInt(AGREED_TO_PRIVACY_POLICY_VERSION, 0);
-		int currentPrivacyPolicyHash = getResources().getString(R.string.dialog_privacy_policy_text)
-						.hashCode();
-		if (oldPrivacyPolicyHash == currentPrivacyPolicyHash) {
+
+		if (oldPrivacyPolicy == Constants.CATROBAT_TERMS_OF_USE_ACCEPTED) {
 			loadContent();
 		} else {
-			setContentView(R.layout.privacy_policy_view);
+			View view = View.inflate(this, R.layout.privacy_policy_view, null);
+
+			TextView termsOfUseUrlTextView = view.findViewById(R.id.dialog_privacy_policy_text_view_url);
+
+			termsOfUseUrlTextView.setMovementMethod(LinkMovementMethod.getInstance());
+
+			String termsOfUseUrlStringText = getString(R.string.main_menu_terms_of_use);
+
+			String termsOfUseUrl = getString(R.string.terms_of_use_link_template, Constants.CATROBAT_TERMS_OF_USE_URL
+							+ Constants.CATROBAT_TERMS_OF_USE_TOKEN_FLAVOR_URL + BuildConfig.FLAVOR
+							+ Constants.CATROBAT_TERMS_OF_USE_TOKEN_VERSION_URL + BuildConfig.VERSION_CODE,
+							termsOfUseUrlStringText);
+
+			termsOfUseUrlTextView.setText(Html.fromHtml(termsOfUseUrl));
+
+			new AlertDialog.Builder(this)
+					.setNegativeButton(R.string.decline, new DialogInterface.OnClickListener() {
```

Comment:
Please use lambda-expressions here. They should be suggested by Android Studio automatically.